app/components/deadline_banner_component.rb

```diff
@@ -0,0 +1,29 @@
+class DeadlineBannerComponent < ViewComponent::Base
+  attr_accessor :flash_empty
+
+  def initialize(flash_empty:)
+    @flash_empty = flash_empty
+  end
+
+  def render?
+    flash_empty
```

Comment:
It's a bit mixy to have this return true and then the component to render nothing?

Suggested refactor:

```

def render?
  flash_empty? && banner_to_show
end
  
private

def banner_to_show
  return :apply_1_deadline if CycleTimetable.show_apply_1_deadline_banner?
  return :apply_2_deadline if CycleTimetable.show_apply_2_deadline_banner?
  return :cycle_closed if CycleTimetable.show_cycle_closed_banner?
end
```

and then switch on `banner_to_show` in the template?